source/type/type.cpp

```diff
@@ -373,10 +374,20 @@ VJ* MetaVehicleJourney::impl_create_vj(const std::string& uri,
     vj_ptr->uri = uri;
     vj_ptr->idx = pt_data.vehicle_journeys.size();
     vj_ptr->realtime_level = level;
-    auto* new_vp = pt_data.get_or_create_validity_pattern(vp);
+    int day_offset = 0;
+    if (!sts.empty()) {
+        day_offset = sts.front().arrival_time / (ndtu::SECONDS_PER_DAY);
+    }
+    ValidityPattern model_new_vp{disrupted_vp};
+    if (day_offset >= 0) {
+        model_new_vp.days <<= size_t(day_offset);
+    } else {
+        model_new_vp.days >>= size_t(-day_offset);
+    }
```

Comment:
directly using a size_t for day_offset should do the trick as everything is unsigned for the moment. And It won't work as if if arrival_time is signed.